aea/aea.py

```diff
@@ -117,13 +118,9 @@ def resources(self) -> Resources:
     @resources.setter
     def resources(self, resources: "Resources") -> None:
         """Set resources."""
+        assert type(resources) is Resources, "Must set resources."
```

Comment:
This is OK, but that implies the question: why don't we check the type also in the constructor?

Reply:
Fair point!